metapool/prep.py

```diff
@@ -842,3 +843,84 @@ def qiita_scrub_name(name):
         the sample name, formatted for qiita
     """
     return re.sub(r'[^0-9a-zA-Z\-\.]+', '.', name)
+
+
+def _organize_by_project(pre_prep):
+    # extract project-level metadata
+
+    gb = pre_prep.groupby('project_name')
+
+    results = [gb.get_group(grp) for grp in gb.groups]
+
+    projects = []
+    data = {}
+
+    for project in results:
```

Comment:
I don't follow why this needs to be split in several lines, it can be:
`for project_name, project in  pre_prep.groupby('project_name'):`

Reply:
Thanks for the tip!